frontend/src/reducers/user.js

```diff
@@ -0,0 +1,140 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const initialState = {
+  login: {
+    accessToken: localStorage.accessToken || null,
+    userName: localStorage.userName || "",
+    secretMessage: null,
+    errorMessage: null,
```

Comment:
You forgot about userID to create it here and in the localStorage, so it is blocking your users from getting a secret message from the server after they refresh the page. 😸